config/http_config.go

```diff
@@ -729,19 +835,20 @@ func (rt *oauth2RoundTripper) RoundTrip(req *http.Request) (*http.Response, erro
 		changed bool
 	)
 
-	if rt.config.ClientSecretFile != "" {
-		data, err := os.ReadFile(rt.config.ClientSecretFile)
+	clientSecret, err := secretFrom(rt.opts.secretManager, rt.config.ClientSecret, rt.config.ClientSecretFile, rt.config.ClientSecretRef)
+	if err != nil {
+		return nil, fmt.Errorf("unable to use client secret: %w", err)
+	}
+	if clientSecret != nil {
+		var err error
+		secret, err = clientSecret.fetch(req.Context())
 		if err != nil {
-			return nil, fmt.Errorf("unable to read oauth2 client secret file %s: %w", rt.config.ClientSecretFile, err)
+			return nil, fmt.Errorf("unable to read oauth2 client secret: %w", err)
 		}
-		secret = strings.TrimSpace(string(data))
-		rt.mtx.RLock()
-		changed = secret != rt.secret
-		rt.mtx.RUnlock()
-	} else {
-		// Either an inline secret or nothing (use an empty string) was provided.
-		secret = string(rt.config.ClientSecret)
 	}
+	rt.mtx.RLock()
```

Comment:
should we leverage immutable method to avoid any lock, similar to prev part:

<img width="651" alt="image" src="https://github.com/prometheus/common/assets/6950331/ddeedfa7-e4cc-4349-b355-7a0168002f20">

Reply:
Plus we can move it to `if clientSecret != nil {` body?